FordLabs/retroquest
# Migrate from Protractor to Cypress

## Overview
- Modify End to End Tests to use Cypress instead of protractor
- Add Testing Library to enable UI testing within Jest
- Update ship.sh script to work
- Update documentation to describe the three layers of testing

Connects #282

## Notes
- Removed Protractor
- Removed e2e configuration in angular.json (except linting)
- End to End tests in cypress are more thorough than before.  Simulates a mini-retro
- The e2e tests in protractor that were actually UI tests have been moved to unit tests using the testing library

## Testing Instructions
After checking out branch, install new dependencies and run all unit tests.  There should be no errors
```
cd ./ui
yarn install
yarn unit
```
Next run cypress in headless mode; all tests should pass
```
yarn cypress
```
Next, run cypress in supervise mode, all tests should pass
```
yarn cypress-supervise
```
